stl/inc/xlocnum

```diff
@@ -670,9 +670,9 @@ private:
         const string _Grouping  = _Punct_fac.grouping();
         const _Elem _Kseparator = _Grouping.size() == 0 ? _Elem{} : _Punct_fac.thousands_sep();
 
-        constexpr int _Numget_signoff             = 22;
-        constexpr int _Numget_xoff                = 24;
-        _STATIC_UNLESS_PURE constexpr char _Src[] = "0123456789ABCDEFabcdef-+Xx";
+        constexpr int _Numget_signoff = 22;

```

Comment:
Should these be static too?

Reply:
(Occurs below)

Reply:
According to our current conventions, we apply `static` to `constexpr` arrays where it makes a significant difference for codegen (`static constexpr` arrays live in the read-only data segment of the binary so the compiler just pushes their address onto the stack; automatic `constexpr` arrays must have distinct addresses so the compiler emits code to push every value onto the stack). This doesn't make a difference for non-array variables that aren't address-taken, which describes the vast majority of our constants (we generally don't pass them by reference, etc.).

We could change our conventions to mark all `constexpr` function-local variables as `static` (and indeed auditing for any missed occurrences on arrays is what led me to this change), but I would want to handle that as a separate `enhancement` issue, not part of this workaround removal.